app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -88,39 +95,100 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
         if (plans.isEmpty()) {
             throw new EntityNotFoundException(SchedulePlan.class);
         }
-        return plans.get(0);
+        
+        plan = plans.get(0);
+        forEachCriteria(plan, scheduleCriteria -> loadCriteria(scheduleCriteria));
+        return plan;
     }
 
     @Override
     public SchedulePlan createSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
-        checkNotNull(studyIdentifier, "StudyIdentifier is null");
-        checkNotNull(plan, "Schedule plan is null");
+        checkNotNull(studyIdentifier);
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setGuid(BridgeUtils.generateGuid());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
```

Comment:
I think we can go further and have a strategy table, but that needs to be incremental. Pulling out criteria simplifies this particular strategy object and it will make a later migration easier. But since DDB doesn't support joins it's also risky to query multiple tables for the data, so this might be better done if/when we move to a RDBMS.